src/convertToObject.js

```diff
@@ -3,16 +3,34 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const propertiesArray = sourceString.split(';');
+
+  return propertiesArray.reduce((style, current) => {
+    const [key, value] = current.split(':');
+    const styleKey = key
+      ? key.trim()
+      : null;
+    const styleValue = value
+      ? value.trim()
+      : null;
+
+    if (!style[styleKey] && styleKey && styleValue) {
```

Comment:
`!style[styleKey] && styleKey && ` remove these conditions. It will work without these)

Reply:
thank you, yes, I made a cheking in ternary and forgot replace it